src/neo-vm/ExecutionEngine.cs

```diff
@@ -42,6 +92,164 @@ public void AddBreakPoint(byte[] script_hash, uint position)
             hashset.Add(position);
         }
 
+        public bool RemoveBreakPoint(byte[] script_hash, uint position)
+        {
+            if (break_points == null)
+                return false;
+            if (!break_points.TryGetValue(script_hash, out HashSet<uint> hashset))
+                return false;
+            if (!hashset.Remove(position))
+                return false;
+            if (hashset.Count == 0)
+                break_points.Remove(script_hash);
+            return true;
+        }
+
+        #endregion
+
+        #region Limits
+
+        /// <summary>
+        /// Check if it is possible to overflow the MaxArraySize
+        /// </summary>
+        /// <param name="length">Length</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckArraySize(int length) => length <= MaxArraySize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxItemSize
+        /// </summary>
+        /// <param name="length">Length</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxItemSize(int length) => length <= MaxItemSize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxInvocationStack
+        /// </summary>
+        /// <param name="stack">Stack</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxInvocationStack(ExecutionEngine stack) => stack.InvocationStack.Count < MaxInvocationStackSize;
```

Comment:
All calls to `CheckMaxInvocationStack()` are passed in the `this` pointer. Why do we need the parameter?

Reply:
Was moved from my previous code, fixed now :) thanks!